src/Commands/RunRepeater.ts

```diff
@@ -149,46 +160,74 @@ export class RunRepeater implements CommandModule {
 
         return true;
       })
-      .middleware((args: Arguments) =>
-        container
-          .register<RequestExecutorOptions>(RequestExecutorOptions, {
-            useValue: {
-              headers: (args.header ?? args.headers) as Record<string, string>,
-              timeout: args.timeout as number,
-              proxyUrl: (args.proxyInternal ?? args.proxy) as string,
-              certs: args.cert as Cert[],
-              maxContentLength: 100,
-              reuseConnection: !!args.experimentalConnectionReuse,
-              whitelistMimes: [
-                'text/html',
-                'text/plain',
-                'text/css',
-                'text/javascript',
-                'text/markdown',
-                'text/xml',
-                'application/javascript',
-                'application/x-javascript',
-                'application/json',
-                'application/xml',
-                'application/x-www-form-urlencoded',
-                'application/msgpack',
-                'application/ld+json',
-                'application/graphql'
-              ]
-            }
-          })
-          .register<DefaultRepeaterServerOptions>(
-            DefaultRepeaterServerOptions,
-            {
+      .middleware((args: Arguments) => {
+        container.register<RequestExecutorOptions>(RequestExecutorOptions, {
+          useValue: {
+            headers: (args.header ?? args.headers) as Record<string, string>,
+            timeout: args.timeout as number,
+            proxyUrl: (args.proxyInternal ?? args.proxy) as string,
+            certs: args.cert as Cert[],
+            maxContentLength: 100,
+            reuseConnection: !!args.experimentalConnectionReuse,
+            whitelistMimes: [
+              'text/html',
+              'text/plain',
+              'text/css',
+              'text/javascript',
+              'text/markdown',
+              'text/xml',
+              'application/javascript',
+              'application/x-javascript',
+              'application/json',
+              'application/xml',
+              'application/x-www-form-urlencoded',
+              'application/msgpack',
+              'application/ld+json',
+              'application/graphql'
+            ]
+          }
+        });
+
+        if (args.legacy) {
```

Comment:
You can register options and other providers as values moving them out from the if/else statement. The if/else can be rewritten for clarity:


```ts
container
  .register<RepeaterLauncher>(
    RepeaterLauncher,
    { useClass: args.legacy ? DefaultRepeaterLauncher : ServerRepeaterLauncher },
    { lifecycle: Lifecycle.Singleton }
  )
```